pallets/nfts/src/lib.rs

```diff
@@ -406,6 +404,46 @@ pub mod pallet {
 	pub type CollectionConfigOf<T: Config<I>, I: 'static = ()> =
 		StorageMap<_, Blake2_128Concat, T::CollectionId, CollectionConfigFor<T, I>, OptionQuery>;
 
+	/// Number of collection items that accounts own.
+	#[pallet::storage]
+	pub type AccountBalance<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		T::CollectionId,
+		Blake2_128Concat,
+		T::AccountId,
+		u32,
```

Comment:
Why should it be T::ItemId? This tracks the total number of items that the account owns in the collection.

Reply:
Because if I configured the pallet with item id of u64, that potentially overflows your u32 balance here if all items are minted to a single account.

Reply:
Note: Explain more why the overflow case can never happen.

Below is the code snippet for the `do_mint` method. 
```rs
	if let Some(max_supply) = collection_config.max_supply {
		ensure!(collection_details.items < max_supply, Error::<T, I>::MaxSupplyReached);
	}

	collection_details.items.saturating_inc();

	AccountBalance::<T, I>::mutate(collection, &mint_to, |balance| {
		balance.saturating_inc();
	});
```
One collection can only reach up to the configured max supply. While the max supply data type is enforced `u32` to 

```
pub max_supply: Option<u32>,
```

In that way, `AccountBalance` can never go pass `u32::Max` because there are only up to `u32::Max` items in the collection so they can't have more items that what are supplied. 

And with `saturating_inc`, this will be likely never throw unexpected error. 

If so, is it fair to say the integration test is no longer needed?

Reply:
Realized `max_supply` can be set to `None` so integration test is still required. Sorry for the noise, I'll leave the above message as additional information.

Reply:
Sounds good! I have no problem with having more protection over the code, so I'll keep the integration test. Thank you!